batch/batch/front_end/front_end.py

```diff
@@ -666,7 +666,12 @@ async def get_job_log(request: web.Request, _, batch_id: int) -> web.Response:
             raise web.HTTPBadRequest(
                 reason=f'log for container {container} is not valid UTF-8, upgrade your hail version to download the log'
             ) from e
-    return json_response(job_log_strings)
+    return json_response(
+        job_log_strings,
+        headers={
+            "X-Hail-Deprecated": 'The endpoint "/api/v1alpha/batches/<batch_id>/jobs/<job_id>/log" is deprecated. Please upgrade your Hail version to the latest release.'
+        },
```

Comment:
How many routes are we likely to deprecate? While this solution is fine I'm not too fond of the repeated route names. 
Would a modified `routes` decorator that automatically adds this deprecation message be too complicated/overly fancy:
```python
@get(route, deprecated=True)
async def get_foo(request: Request) -> Response: 
   ...


def get(path, deprecated: bool): 
    return route(routes.get, path, deprecated)

def post(path, deprecated):
    return route(routes.post, path, deprecated)

...

def route(verb, path, deprecated):
    def wrapper(fn):
        @verb(path)
        async def handler(request, *args, **kwargs):
            response = await fn(request, args, kwards)
            if deprecated:
                response.headers['X-Hail-Deprecated'] = (
                    f'The endpoint "{path}" is deprecated. '
                    'Please upgrade your Hail version to the latest release.'
                )
           return response
        return handler
    return wapper

```

Reply:
I think that's fine if Iris wants to do it, but I expect deprecated endpoints to be pretty rare.